service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CopyGcsBucketDataStep.java

```diff
@@ -0,0 +1,235 @@
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, e);
+    } else {
+      logger.debug("Completed operation metadata: {}", operation.getMetadata());
+      return StepResult.getStepResultSuccess();
+    }
+  }
+
+  // First, we poll the transfer jobs endpoint until an operation has started so that we can get
+  // its server-generated name. Returns the most recently started operation's name.
+  private String getLatestOperationName(
+      Storagetransfer storageTransferService, String transferJobName, String projectId)
+      throws InterruptedException, IOException {
+    int attempts = 0;
+    String operationName;
+    do {
+      TimeUnit.MILLISECONDS.sleep(JOBS_POLL_INTERVAL.toMillis());
+      final TransferJob getResponse =
+          storageTransferService.transferJobs().get(transferJobName, projectId).execute();
+      operationName = getResponse.getLatestOperationName();
+      attempts++;
+      if (attempts > MAX_ATTEMPTS) {
+        throw new RuntimeException("Exceeded max attempts to get transfer operation name");
+      }
+    } while (operationName == null);
+    logger.info("Latest transfer operation name is {}", operationName);
+    return operationName;
+  }
+
+  private TransferSpec createTransferSpec(String sourceBucketName, String destinationBucketName) {
+    return new TransferSpec()
+        .setGcsDataSource(new GcsData().setBucketName(sourceBucketName))
+        .setGcsDataSink(new GcsData().setBucketName(destinationBucketName))
+        .setTransferOptions(
+            new TransferOptions()
+                .setDeleteObjectsFromSourceAfterTransfer(false)
+                .setOverwriteObjectsAlreadyExistingInSink(false));
+  }
+
+  /**
+   * Build a schedule to indicate that the job should run an operation immediately and not repeat
+   * it. This isn't well-supported in the UI, but there are hints in the documentation. The most
+   * frustrating feature of the interface is that if a start date & time are in "the past", the run
+   * won't initiate until the next iteration (e.g. in 24 hours, assuming that's not past the end
+   * time). Which means we need to apply a fudge factor to make sure the time is a little bit in the
+   * future. Note that stepping through the code in the debugger too slowly can leave you with a
+   * stale timestamp and cause the job not to run.
+   *
+   * @return schedule object for the transfer job
+   */
+  private Schedule createScheduleRunOnceNow() {
+    final OffsetDateTime now = OffsetDateTime.now(ZoneOffset.UTC);
+    final Date runDate =
+        new Date().setYear(now.getYear()).setMonth(now.getMonthValue()).setDay(now.getDayOfMonth());
+    // "If `schedule_end_date` and schedule_start_date are the same and
+    // in the future relative to UTC, the transfer is executed only one time."
+    return new Schedule().setScheduleStartDate(runDate).setScheduleEndDate(runDate);
```

Comment:
I think this conflicts a bit with this methods javadoc - is this setting the time "a little bit in the future"? it looks like we're not setting the start time at all, and I can't tell what that does from GCP documentation

Reply:
good catch. it's an old comment. I _think_ "future relative to UTC" means "today or later". 

With no start time, it starts  immediately and runs once, from my reading of the javadoc.